File: SpingBoot/Framewoork/BackCustomer/src/main/java/com/Backend3/BackCustomer/Service/CustomerService.java
package com.Backend3.BackCustomer.Service;

import com.Backend3.BackCustomer.Model.Customer;

public interface CustomerService {

    Iterable<Customer> getList();
    Customer getCustomer(int id);
    void DeleteCustomer(int id);
    Customer UpdateCustomer(int id,Customer newCustomer);
    Customer CrearCustomer(Customer newCustomer);
    Customer StartWith(String nombre);
}

File: Angular/Fronted1/src/app/service/customer.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Customer } from '../model/customer';
import { CustomerSinID } from '../model/customer-sin-id';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  
  private api:string="http://localhost:8080/api/Customer";

  constructor(private http:HttpClient) {}

  getCustomerList():Observable<Customer[]>  {
    return this.http.get<Customer[]>(this.api);
  }

  createCustomer(newCustomer:CustomerSinID):Observable<CustomerSinID>{
    return this.http.post<CustomerSinID>(this.api,newCustomer);
  }

  DeleteCustomer(id:number):Observable<any>{
    return this.http.delete(this.api+"/"+id);
  }

  UpdateCustomer(newCustomer:Customer):Observable<Customer>{
    return this.http.put<Customer>(this.api , newCustomer)
  }

  fetchByID(id:number):Observable<Customer>{
    return this.http.get<Customer>(this.api+"/"+id)
  }

}
